### DISCUSSION TITLE
Install on TrueNas scale

### DISCUSSION THREAD

**Discussion by ozzy071988** [d3vilh/openvpn-ui#57] [Q&A] [answered]
Hi every one.

Has anyone tried installing this on truenas scale? I'm struggling with this but I can't cope

**Comment by d3vilh** [comment]
Hi @ozzy071988 
As I remember TrueNAS is FreeBSD😈 based, technically you could run Docker env in VM and on this VM have everything running (depending on cpu arch).
I would recommend you start with VM and docker-compose to try running container based on existent images (we have x86_64, ARMv8(64) and ARM6 (32 bit)). 

If you don't want to deal with Docker, then you could try `standalone-install.sh` which will build all the binaries for you (it is depending for GoLang). Don't run it as it is, it currently supports Debian only, so should be updated to support ports and other things. But technically - yes you could build it and it will run without any problem natively on FreeBSD.

Let me know your `uname -a` output, just to confirm I have containers build for your HW+SW.

**Reply by ozzy071988** [reply, marked as answer]
Hi. Thank u for reply.

This is output from uname -a

Linux truenas 6.1.74-production+truenas #2 SMP PREEMPT_DYNAMIC Wed Feb 21 20:30:38 UTC 2024 x86_64 GNU/Linux

**Reply by d3vilh** [reply]
OK, you could try to run OpenVPN-UI as Docker container with TrueNas App. 
Here are [short manual for this](https://www.wundertech.net/how-to-use-docker-on-truenas-scale/). 

Image is `d3vilh/openvpn-ui:latest` 
Rest parameters, as ports and volumes can be taken from [docker compose file](https://github.com/d3vilh/openvpn-ui?tab=readme-ov-file#manual-installation).

It should run, as you have x86_64 arch.
